File: src/main/java/team4/footwithme/vote/service/response/AllVoteResponse.java
package team4.footwithme.vote.service.response;

import team4.footwithme.vote.domain.Vote;

import java.time.LocalDateTime;

public record AllVoteResponse(
    Long voteId,
    String title,
    LocalDateTime endAt,
    String status
) {
    public static AllVoteResponse from(Vote vote) {
        return new AllVoteResponse(
            vote.getVoteId(),
            vote.getTitle(),
            vote.getEndAt(),
            vote.getVoteStatus().getText()
        );
    }
}

File: src/main/java/team4/footwithme/vote/service/response/AllVoteResponse.kt
package team4.footwithme.vote.service.response

import team4.footwithme.vote.domain.Vote
import java.time.LocalDateTime

@JvmRecord
data class AllVoteResponse(
    val voteId: Long?,
    val title: String?,
    val endAt: LocalDateTime?,
    val status: String?
) {
    companion object {
        fun from(vote: Vote?): AllVoteResponse {
            return AllVoteResponse(
                vote!!.voteId,
                vote.title,
                vote.endAt,
                vote.voteStatus!!.text
            )
        }
    }
}
